python/cudf/cudf/core/dataframe.py

```diff
@@ -825,19 +824,19 @@ def astype(self, dtype, copy=True, errors="raise", **kwargs):
         result = self.copy(deep=copy)
```

Comment:
This will possibly deep copy just to copy the columns that get typecasted again.

Reply:
Fixed by constructing a dataframe and letting the downstream `astype` make decisions based on `copy` parameter.